File: app-macos/Sources/RunnerForge/Models/PreflightCheck.swift
import Foundation

/// Outcome of one preflight check.
public enum PreflightStatus: String, Sendable {
    case checking
    case pass
    /// Not met, but only some runner classes are affected.
    case warn
    /// Not met, and the affected classes cannot run at all.
    case fail
}

/// One row on the Preflight page: what was checked, what was found, and whether
/// Runner Forge can fix it.
public struct PreflightCheck: Sendable, Identifiable, Hashable {
    public let id = UUID()
    public let name: String
    public let status: PreflightStatus

    /// What was actually found. Concrete, not "something went wrong".
    public let detail: String

    /// What the user (or the Fix button) should do. Nil when there is nothing to do.
    public let fixHint: String?

    /// True when the Fix button can resolve this without leaving the app.
    public let autoFixable: Bool

    /// Runner classes this check blocks, by classId. Empty when it blocks nothing.
    public let blocksClasses: [String]

    /// True when the failure has no workaround at all — an Intel Mac, for
    /// instance. The UI states the reason rather than offering a Fix that cannot
    /// possibly work.
    public let isHardBlock: Bool

    public init(
        name: String,
        status: PreflightStatus,
        detail: String,
        fixHint: String? = nil,
        autoFixable: Bool = false,
        blocksClasses: [String] = [],
        isHardBlock: Bool = false
    ) {
        self.name = name
        self.status = status
        self.detail = detail
        self.fixHint = fixHint
        self.autoFixable = autoFixable
        self.blocksClasses = blocksClasses
        self.isHardBlock = isHardBlock
    }
}

File: app-windows/RunnerForge/Models/PreflightCheck.cs
namespace RunnerForge.Models;

/// <summary>Outcome of one preflight check.</summary>
public enum PreflightStatus
{
    Checking,
    Pass,

    /// <summary>Not met, but only some runner classes are affected.</summary>
    Warn,

    /// <summary>Not met, and the affected classes cannot run at all.</summary>
    Fail,
}

/// <summary>
/// One row on the Preflight page: what was checked, what was found, and whether
/// Runner Forge can fix it.
/// </summary>
public sealed record PreflightCheck
{
    public required string Name { get; init; }

    public required PreflightStatus Status { get; init; }

    /// <summary>What was actually found. Concrete, not "something went wrong".</summary>
    public required string Detail { get; init; }

    /// <summary>What the user (or the Fix button) should do. Null when nothing to do.</summary>
    public string? FixHint { get; init; }

    /// <summary>True when the Fix button can resolve this without the user leaving the app.</summary>
    public bool AutoFixable { get; init; }

    /// <summary>Runner classes this check blocks, by classId. Empty when it blocks nothing.</summary>
    public IReadOnlyList<string> BlocksClasses { get; init; } = [];

    /// <summary>
    /// True when this failure has no workaround at all — Windows Home, or an
    /// Intel Mac. The UI states the reason rather than offering a Fix that
    /// cannot work.
    /// </summary>
    public bool IsHardBlock { get; init; }

    public bool RequiresReboot { get; init; }
}
